### ISSUE TITLE
PHP8 Rewrite

### ISSUE THREAD

**Issue by Dygear** [phpLiteAdmin/pla#31]
I've made a version of [phpLiteAdmin](https://github.com/Dygear/PHPLiteAdmin/commit/9114b90837ae7fe3c9a0ee47557f45af008fafdc) basing it off a PHPLiteAdmin that I made 12 years ago. Key features include:

- It's a complete rewrite of the software for use with PHP 8.0.
- It uses only 1 file, just like SQLite databases themselves.
- The file itself is self mutable to update databases and users it tracks from within the web interface.

Please have a look and let me know what you think.

**Comment by Hope-IT-Works** [comment]
Hey,

thanks for working on this!

Took me 30 minutes to realize that my PHP version was too high 😄 

Is it a whole different architecture or do you try to build it like it was before?

**Comment by crazy4chrissi** [comment]
> Took me 30 minutes to realize that my PHP version was too high smile

The normal  development version of phpLiteAdmin works with PHP 8.

https://www.phpliteadmin.org/phpliteadmin-dev.zip

**Comment by Dygear** [comment]
> Hey,
> 
> thanks for working on this!
> 
> Took me 30 minutes to realize that my PHP version was too high 😄 
> 
> Is it a whole different architecture or do you try to build it like it was before?

Same concept yes. But all code is new. More than happy to release and sign what ever is needed to to the phpLiteAdmin dev team to get this merged.

**Comment by quantuumsnot** [comment]
Fast report ...
This rewrite seems to work faster but there's a problem - created a new user, logged in with its credentials, added already existing database and the script appended it into 

```
$Users = [
];
```
instead in

```
$Database = [
];
```
OK, fixed that adding it manually but checking the data in one of tables the UI doesn't show next page of rows

**Comment by Dygear** [comment]
Yeah, that's something I still haven't delt with (pages) but I'll get around to it. Didn't notice the problem with $database / $users. Wonder how I messed that up. (Probably just copied the code from one place to another.)

**Comment by quantuumsnot** [comment]
Sometimes people do harm with admin panels like this one, so can you add the following code for query logging in 2.0?

The function that logs the query
```
function logQuery($queryforsave) {
  $fileName = "logs/" . date('d M Y') . ".txt";
  file_put_contents($fileName, $queryforsave, FILE_APPEND);
}
```

Example of what is passed to logQuery() function (from my small sales and inventory control system)
```
$queryforsave = "Product was sold in {$soldIn} >> UPDATE products SET quantity=quantity-{$_POST['sellquantity']} WHERE number={$sellNumber}" . PHP_EOL;
 logQuery($queryforsave);
```

Just add the user doing this query, or change whatever you'll find is ok

**Comment by Dygear** [comment]
Hard to make that a generalized thing. I don't know that everyone would want logging, but I could understand why some would. Perhaps I need an $options array that enables disables logging for those that want it.
